File: javascript/src/model/PagingInfo.js
(function(root, factory) {
  if (typeof define === 'function' && define.amd) {
    // AMD. Register as an anonymous module.
    define(['../ApiClient'], factory);
  } else if (typeof module === 'object' && module.exports) {
    // CommonJS-like environments that support module.exports, like Node.
    module.exports = factory(require('../ApiClient'));
  } else {
    // Browser globals (root is window)
    if (!root.RingCentralApi) {
      root.RingCentralApi = {};
    }
    root.RingCentralApi.PagingInfo = factory(root.RingCentralApi.ApiClient);
  }
}(this, function(ApiClient) {
  'use strict';

  /**
   * The PagingInfo model module.
   * @module model/PagingInfo
   * @version v1.0
   */

  /**
   * Constructs a new <code>PagingInfo</code>.
   * @alias module:model/PagingInfo
   * @class
   */
  var exports = function() {







  };

  /**
   * Constructs a <code>PagingInfo</code> from a plain JavaScript object, optionally creating a new instance.
   * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
   * @param {Object} data The plain JavaScript object bearing properties of interest.
   * @param {module:model/PagingInfo} obj Optional instance to populate.
   * @return {module:model/PagingInfo} The populated <code>PagingInfo</code> instance.
   */
  exports.constructFromObject = function(data, obj) {
    if (data) { 
      obj = obj || new exports();

      if (data.hasOwnProperty('page')) {
        obj['page'] = ApiClient.convertToType(data['page'], 'Integer');
      }
      if (data.hasOwnProperty('perPage')) {
        obj['perPage'] = ApiClient.convertToType(data['perPage'], 'Integer');
      }
      if (data.hasOwnProperty('pageStart')) {
        obj['pageStart'] = ApiClient.convertToType(data['pageStart'], 'Integer');
      }
      if (data.hasOwnProperty('pageEnd')) {
        obj['pageEnd'] = ApiClient.convertToType(data['pageEnd'], 'Integer');
      }
      if (data.hasOwnProperty('totalPages')) {
        obj['totalPages'] = ApiClient.convertToType(data['totalPages'], 'Integer');
      }
      if (data.hasOwnProperty('totalElements')) {
        obj['totalElements'] = ApiClient.convertToType(data['totalElements'], 'Integer');
      }
    }
    return obj;
  }


  /**
   * The current page number. 1-indexed, so the first page is 1 by default. May be omitted if result is empty (because non-existent page was specified or perPage=0 was requested)
   * @member {Integer} page
   */
  exports.prototype['page'] = undefined;

  /**
   * Current page size, describes how many items are in each page. Default value is 100. Maximum value is 1000. If perPage value in the request is greater than 1000, the maximum value (1000) is applied
   * @member {Integer} perPage
   */
  exports.prototype['perPage'] = undefined;

  /**
   * The zero-based number of the first element on the current page. Omitted if the page is omitted or result is empty
   * @member {Integer} pageStart
   */
  exports.prototype['pageStart'] = undefined;

  /**
   * The zero-based index of the last element on the current page. Omitted if the page is omitted or result is empty
   * @member {Integer} pageEnd
   */
  exports.prototype['pageEnd'] = undefined;

  /**
   * The total number of pages in a dataset. May be omitted for some resources due to performance reasons
   * @member {Integer} totalPages
   */
  exports.prototype['totalPages'] = undefined;

  /**
   * The total number of elements in a dataset. May be omitted for some resource due to performance reasons
   * @member {Integer} totalElements
   */
  exports.prototype['totalElements'] = undefined;




  return exports;
}));

File: csharp/src/main/csharp/IO/Swagger/Model/PagingInfo.cs
using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace IO.Swagger.Model
{
    /// <summary>
    /// 
    /// </summary>
    [DataContract]
    public partial class PagingInfo :  IEquatable<PagingInfo>
    { 
    
        /// <summary>
        /// Initializes a new instance of the <see cref="PagingInfo" /> class.
        /// Initializes a new instance of the <see cref="PagingInfo" />class.
        /// </summary>
        /// <param name="Page">The current page number. 1-indexed, so the first page is 1 by default. May be omitted if result is empty (because non-existent page was specified or perPage=0 was requested).</param>
        /// <param name="PerPage">Current page size, describes how many items are in each page. Default value is 100. Maximum value is 1000. If perPage value in the request is greater than 1000, the maximum value (1000) is applied.</param>
        /// <param name="PageStart">The zero-based number of the first element on the current page. Omitted if the page is omitted or result is empty.</param>
        /// <param name="PageEnd">The zero-based index of the last element on the current page. Omitted if the page is omitted or result is empty.</param>
        /// <param name="TotalPages">The total number of pages in a dataset. May be omitted for some resources due to performance reasons.</param>
        /// <param name="TotalElements">The total number of elements in a dataset. May be omitted for some resource due to performance reasons.</param>

        public PagingInfo(int? Page = null, int? PerPage = null, int? PageStart = null, int? PageEnd = null, int? TotalPages = null, int? TotalElements = null)
        {
            this.Page = Page;
            this.PerPage = PerPage;
            this.PageStart = PageStart;
            this.PageEnd = PageEnd;
            this.TotalPages = TotalPages;
            this.TotalElements = TotalElements;
            
        }
        
    
        /// <summary>
        /// The current page number. 1-indexed, so the first page is 1 by default. May be omitted if result is empty (because non-existent page was specified or perPage=0 was requested)
        /// </summary>
        /// <value>The current page number. 1-indexed, so the first page is 1 by default. May be omitted if result is empty (because non-existent page was specified or perPage=0 was requested)</value>
        [DataMember(Name="page", EmitDefaultValue=false)]
        public int? Page { get; set; }
    
        /// <summary>
        /// Current page size, describes how many items are in each page. Default value is 100. Maximum value is 1000. If perPage value in the request is greater than 1000, the maximum value (1000) is applied
        /// </summary>
        /// <value>Current page size, describes how many items are in each page. Default value is 100. Maximum value is 1000. If perPage value in the request is greater than 1000, the maximum value (1000) is applied</value>
        [DataMember(Name="perPage", EmitDefaultValue=false)]
        public int? PerPage { get; set; }
    
        /// <summary>
        /// The zero-based number of the first element on the current page. Omitted if the page is omitted or result is empty
        /// </summary>
        /// <value>The zero-based number of the first element on the current page. Omitted if the page is omitted or result is empty</value>
        [DataMember(Name="pageStart", EmitDefaultValue=false)]
        public int? PageStart { get; set; }
    
        /// <summary>
        /// The zero-based index of the last element on the current page. Omitted if the page is omitted or result is empty
        /// </summary>
        /// <value>The zero-based index of the last element on the current page. Omitted if the page is omitted or result is empty</value>
        [DataMember(Name="pageEnd", EmitDefaultValue=false)]
        public int? PageEnd { get; set; }
    
        /// <summary>
        /// The total number of pages in a dataset. May be omitted for some resources due to performance reasons
        /// </summary>
        /// <value>The total number of pages in a dataset. May be omitted for some resources due to performance reasons</value>
        [DataMember(Name="totalPages", EmitDefaultValue=false)]
        public int? TotalPages { get; set; }
    
        /// <summary>
        /// The total number of elements in a dataset. May be omitted for some resource due to performance reasons
        /// </summary>
        /// <value>The total number of elements in a dataset. May be omitted for some resource due to performance reasons</value>
        [DataMember(Name="totalElements", EmitDefaultValue=false)]
        public int? TotalElements { get; set; }
    
        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append("class PagingInfo {\n");
            sb.Append("  Page: ").Append(Page).Append("\n");
            sb.Append("  PerPage: ").Append(PerPage).Append("\n");
            sb.Append("  PageStart: ").Append(PageStart).Append("\n");
            sb.Append("  PageEnd: ").Append(PageEnd).Append("\n");
            sb.Append("  TotalPages: ").Append(TotalPages).Append("\n");
            sb.Append("  TotalElements: ").Append(TotalElements).Append("\n");
            
            sb.Append("}\n");
            return sb.ToString();
        }
  
        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public string ToJson()
        {
            return JsonConvert.SerializeObject(this, Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="obj">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object obj)
        {
            // credit: http://stackoverflow.com/a/10454552/677735
            return this.Equals(obj as PagingInfo);
        }

        /// <summary>
        /// Returns true if PagingInfo instances are equal
        /// </summary>
        /// <param name="other">Instance of PagingInfo to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(PagingInfo other)
        {
            // credit: http://stackoverflow.com/a/10454552/677735
            if (other == null)
                return false;

            return 
                (
                    this.Page == other.Page ||
                    this.Page != null &&
                    this.Page.Equals(other.Page)
                ) && 
                (
                    this.PerPage == other.PerPage ||
                    this.PerPage != null &&
                    this.PerPage.Equals(other.PerPage)
                ) && 
                (
                    this.PageStart == other.PageStart ||
                    this.PageStart != null &&
                    this.PageStart.Equals(other.PageStart)
                ) && 
                (
                    this.PageEnd == other.PageEnd ||
                    this.PageEnd != null &&
                    this.PageEnd.Equals(other.PageEnd)
                ) && 
                (
                    this.TotalPages == other.TotalPages ||
                    this.TotalPages != null &&
                    this.TotalPages.Equals(other.TotalPages)
                ) && 
                (
                    this.TotalElements == other.TotalElements ||
                    this.TotalElements != null &&
                    this.TotalElements.Equals(other.TotalElements)
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            // credit: http://stackoverflow.com/a/263416/677735
            unchecked // Overflow is fine, just wrap
            {
                int hash = 41;
                // Suitable nullity checks etc, of course :)
                
                if (this.Page != null)
                    hash = hash * 59 + this.Page.GetHashCode();
                
                if (this.PerPage != null)
                    hash = hash * 59 + this.PerPage.GetHashCode();
                
                if (this.PageStart != null)
                    hash = hash * 59 + this.PageStart.GetHashCode();
                
                if (this.PageEnd != null)
                    hash = hash * 59 + this.PageEnd.GetHashCode();
                
                if (this.TotalPages != null)
                    hash = hash * 59 + this.TotalPages.GetHashCode();
                
                if (this.TotalElements != null)
                    hash = hash * 59 + this.TotalElements.GetHashCode();
                
                return hash;
            }
        }

    }
}
